template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Areas/IdentityService/IdentityServiceExtensions.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Identity.EntityFrameworkCore;
+using Microsoft.AspNetCore.Identity.Service;
+using Microsoft.AspNetCore.Identity.Service.EntityFrameworkCore;
+using Microsoft.AspNetCore.Identity.Service.IntegratedWebClient;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Company.WebApplication1.Identity.Data;
+using Company.WebApplication1.Identity;
+using Company.WebApplication1.Identity.Models;
+using Company.WebApplication1.Identity.Services;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class IdentityServiceExtensions
+    {
+        public static IServiceCollection AddIdentityService(this IServiceCollection services, IConfiguration config)
+        {
+            // Add framework services.
+            services.AddDbContext<IdentityServiceDbContext>(options =>
+  #if (UseLocalDB)
+                options.UseSqlServer(config.GetConnectionString("DefaultConnection")));
+  #else
+                options.UseSqlite(config.GetConnectionString("DefaultConnection")));
+  #endif
+
+            services.AddIdentity<ApplicationUser, IdentityRole>()
+                .AddEntityFrameworkStores<IdentityServiceDbContext>()
+                .AddDefaultTokenProviders();
+
+            services.AddIdentityService<ApplicationUser, IdentityServiceApplication>(
+                options =>
+                {
+                    options.IdTokenOptions.ContextClaims.AddSingle("tfp", "policy");
+                    options.IdTokenOptions.ContextClaims.AddSingle("ver", "version");
+                    options.AccessTokenOptions.ContextClaims.AddSingle("tfp", "policy");
+                    options.AccessTokenOptions.ContextClaims.AddSingle("ver", "version");
+
+                    options.Issuer = config["Authentication:Identity:Issuer"];
+                })
+                .AddEntityFrameworkStores<IdentityServiceDbContext>()
+                .AddSigningCertificates(() => CertificateLoader.LoadCertificates(
+                    config.GetSection("Authentication:Identity:SigningCertificates")));
+
+            services.AddScoped<IAuthorizationResponseParameterProvider, ClientInfoProvider>();
+            services.AddScoped<ITokenResponseParameterProvider, ClientInfoProvider>();
+
+            // Add external authentication handlers below. To configure them please see https://go.microsoft.com/fwlink/?LinkID=532715
+
+            services.AddIntegratedWebClient(config.GetSection("Authentication:Identity"));
+
+            services.AddTransient<IEmailSender, AuthMessageSender>();
```

Comment:
We should discuss. For now I think we should leave them in. There's quite a bit of wire up code that you would have to add back if we remove these.